converters/google/convert.go

```diff
@@ -212,17 +278,26 @@ func (c *Converter) addCreateOrUpdate(rc tfplan.ResourceChange) error {
 
 		key := converted.Type + converted.Name
 
+		var existingConverterAsset *converter.Asset
 		if existing, exists := c.assets[key]; exists {
-			// The existance of a merge function signals that this tf resource maps to a
-			// patching operation on an API resource.
-			if mapper.mergeCreateUpdate != nil {
-				converted = mapper.mergeCreateUpdate(existing.converterAsset, converted)
-			} else {
+			existingConverterAsset = &existing.converterAsset
+		} else if mapper.fetch != nil{
+			asset, err := mapper.fetch(&rd, c.cfg)
+			if err != nil {
+				return errors.Wrap(err, "fetching asset")
+			}
+
+			existingConverterAsset = &asset
+		}
+
+		if existingConverterAsset != nil {
+			if mapper.mergeCreateUpdate == nil {
 				// If a merge function does not exist ignore the asset and return
 				// a checkable error.
 				return errors.Wrapf(ErrDuplicateAsset, "asset type %s: asset name %s",
 					converted.Type, converted.Name)
```

Comment:
```suggestion
				return fmt.Errorf("asset type %s: asset name %s %w", converted.Type, converted.Name, ErrDuplicateAsset)
```

Reply:
ah, cool, wasn't sure if that would work 👍